code/modules/lighting/lighting_corner.dm

```diff
@@ -192,3 +192,40 @@
 		SSlighting.corners_queue -= src
 
 	return ..()
+
+/datum/lighting_corner/proc/display(max_lum)
+	if(QDELETED(src))
+		return
+
+	var/turf/draw_to = master_SW || master_NE || master_SE || master_NW
+	var/mutable_appearance/display = mutable_appearance('icons/turf/debug.dmi', "corner_color", LIGHT_DEBUG_LAYER, draw_to, BALLOON_CHAT_PLANE)
+	if(x > draw_to.x)
+		display.pixel_x = 16
+	else
+		display.pixel_x = -16
+	if(y > draw_to.y)
+		display.pixel_y = 16
+	else
+		display.pixel_y = -16
+
+	var/index = largest_color_luminosity / max_lum
+	display.color = gradient(0, COLOR_BLACK, 1, COLOR_WHITE, index)
+
+	draw_to.add_overlay(display)
+
+/datum/lighting_corner/dummy/display()
+	return
+
+/proc/display_corners()
+	var/list/corners = list()
+	var/max_lum = 0
+	for(var/datum/lighting_corner/corner) // I am so sorry
+		corners += corner
+		max_lum = max(max_lum, corner.largest_color_luminosity)
+
+
+	for(var/datum/lighting_corner/corner as anything in corners)
+		corner.display(max_lum)
```

Comment:
redundant, need to remove I think? It's a debug tool I was making to help visualize lighting corners, may keep it as a mapping proc just for the sake of it? unsure.